dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/JdbcEventStore.java

```diff
@@ -677,6 +678,17 @@ yield new StringBuilder()
     return query.toString();
   }
 
+  private Double validateAndParseNumericFilter(String context, String uid, String filterValue) {
```

Comment:
Couldn't we catch this 400 earlier in the flow like in `EventOperationParamsMapper`?
Or even in the `filterBy` builder method itself in `EventQueryParams`.
We already have all the information to validate it there.